src/sinol_make/commands/run/__init__.py

```diff
@@ -202,41 +204,82 @@ def execute(self, execution):
 								self.extract_test_no(test)+".res")
 		hard_time_limit_in_s = math.ceil(2*time_limit / 1000.0)
 
-		command = "MEM_LIMIT=%sK MEASURE_MEM=true timeout -k %ds -s SIGKILL %ds %s %s <%s >%s 2>%s" \
-				% (math.ceil(memory_limit), hard_time_limit_in_s,
-					hard_time_limit_in_s, timetool_path,
-					executable, test, output_file, result_file)
-		code = os.system(command)
-		result = {}
-		with open(result_file) as r:
-			for line in r:
-				line = line.strip()
-				if ": " in line:
-					(key, value) = line.split(": ")[:2]
-					result[key] = value
-		if "Time" in result.keys():
-			result["Time"] = self.parse_time(result["Time"])
-		if "Memory" in result.keys():
-			result["Memory"] = self.parse_memory(result["Memory"])
-		if code == 35072:
-			result["Status"] = "TL"
-		elif "Time" in result.keys() and result["Time"] > time_limit:
-			result["Status"] = "TL"
-		elif "Memory" in result.keys() and result["Memory"] > memory_limit:
-			result["Status"] = "ML"
-		elif "Status" not in result.keys():
-			result["Status"] = "RE"
-		elif result["Status"] == "OK":
-			if result["Time"] > time_limit:
+		if self.args.time_tool == 'oiejq':
```

Comment:
Maybe it would be better to split the time vs oiejq `if` to separate documented functions? Currently it's a bit long and it's hard to keep track of what is being changed and what are the assumptions of those ifs.

Reply:
Splitted parsing results into different functions: 09ad06b71c4941db2439cfaabd9116b51c9f2873